python/cudf/cudf/core/frame.py

```diff
@@ -117,7 +139,8 @@ def cast_cols_to_common_dtypes(
                 for cols in list_of_columns:
                     # If column not in this df, fill with an all-null column
                     if idx >= len(cols) or cols[idx] is None:
-                        n = len(next(filter(lambda x: x is not None, cols)))
+                        filtered_cols = [x for x in cols if x is not None]
+                        n = len(filtered_cols[0])
```

Comment:
Why are we changing this? Using `filter` will be much faster as it won't have to operate on each of `cols`:

```python
In [35]: a = list(range(1_000_000))

In [36]: %timeit [x for x in a if x == 1][0]
20.1 ms ± 2.99 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [37]: %timeit next(filter(lambda x: x == 1, a))
288 ns ± 0.252 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
We could alternatively use a generator expression, which is probably the cleanest:

```
next(x for x in a if x == 1)
```

Reply:
Nice, change it to the generator expression. 

Initially had to make this change for the partial empty case where we were getting errors but now it is not necessary.